### ISSUE TITLE
Qtconsole Unicode repr mojibake

### ISSUE THREAD

**Issue by xflr6** [ipython/ipython#8259]
IPython 3.1.0 Qtconsole under Python 2.7.9 (win32):
```python
In [1]: import pandas
In [2]: pandas.DataFrame({'spam': [u'sp\xe4m']})
Out[2]: 
   spam
0  spÃ¤m
```
As opposed to the plain IPython shell:
```python
In [2]: pandas.DataFrame({'spam': [u'sp\xe4m']})
Out[2]:
   spam
0  späm
```

**Comment by ccordoba12** [comment]
I thought this problem was fixed precisely in 3.1.

**Comment by minrk** [comment]
It's a cp1252 vs utf8 problem. Since the encoding of `repr(frame)` bytes is ambiguous, we have to guess. Currently, IPython is guessing with `platform.get_preferred_encoding()`, which is returning cp1252. However, pandas is cleverly asking sys.stdout what its encoding is, and encoding with utf8. Some irony is that the utf8 information is coming from IPython itself.

I think the right behavior here is to change the encoding IPython uses to decode to `sys.stdout.encoding`, which will be utf8 in the qtconsole, because that will be the right answer for any code smart enough to ask the output streams what their encoding should be.

PR #8260 does exactly this.
